### DISCUSSION TITLE
Enforcing landscape fullscreen

### DISCUSSION THREAD

**Discussion by vstreame** [muxinc/media-chrome#487] [Proposals]
Hi all! We really love using media-chrome, but one of the things we we were thinking about was having it so that going full screen on a mobile device would force landscape mode. But it doesn't seem media-chome currently provides the hooks to do this?

I think it would be neat if the `media-controller` or `media-fullscreen-button` took an attribute like, `forceLandscapeFullscreen` which would then ensure the device is changed to landscape during the fullscreen transition. Would that be something that would be interesting to `media-chrome`? 

We would be happy to take a crack at a PR if so. Looks like we could do something similar to how this VideoJS plugin works: https://github.com/prateekrastogi/videojs-landscape-fullscreen/blob/master/src/plugin.js

Cheers and thanks for such a great project!

**Comment by heff** [comment]
Welcome to the project! Great to hear it's working well for you all. This feature idea make sense to me, so if you're up for making a PR, please do!

It should really be a media-controller attribute so that ultimately anything that _requests fullscreen_ would benefit from this.

Feel free to push a draft PR as soon as you get anywhere and ping us with questions as you go. We have channel in [video-dev slack](https://video-dev.fly.dev/) if that's helpful.

**Reply by vstreame** [reply]
Will do!
